### PR TITLE
Feature/3644 remove rest batch infrastructure

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #3644 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Core/REST_API/REST_Routes.php [hunk 1]
```diff
@@ -167,69 +167,6 @@ private function get_routes() {
 		};
 
 		$routes = array(
-			// TODO: This route is super-complex to use and needs to be simplified.
```
Comment (eugene-manuilov) [inline includes/Core/REST_API/REST_Routes.php:170]: We also need to do this:

> * Edit `REST_Routes` class:
>   * **Delete the `error_to_response` method.**

File: includes/Modules/Analytics_4.php [hunk 2]
```diff
@@ -620,6 +615,26 @@ public static function filter_webdatastream_with_ids( $webdatastream ) {
...
+	 * Parses a response, adding the _id and _propertyID params and converting to an array keyed by the propertyID and web datastream IDs.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param GoogleAnalyticsAdminV1alphaListWebDataStreamsResponse[] $response Array of GoogleAnalyticsAdminV1alphaListWebDataStreamsResponse objects.
+	 * @return \stdClass[] Array of models containing _id and _propertyID attributes, keyed by the propertyID.
+	 */
+	public static function filter_webdatastreams_batch( $response ) {
+		$mapped         = array();
+		$results        = array_shift( $response );
```
Comment (eugene-manuilov) [inline includes/Modules/Analytics_4.php:628]: There might be more than 1 response in the batch. Thus we need to iterate over each response in the batch, not just the first one.

File: includes/Modules/Analytics_4.php [hunk 3]
```diff
@@ -620,6 +615,26 @@ public static function filter_webdatastream_with_ids( $webdatastream ) {
...
+		$webdatastreams = $results->getWebDataStreams();
+		foreach ( $webdatastreams as $webdatastream ) {
+			$value          = self::filter_webdatastream_with_ids( $webdatastream );
+			$key            = $value->_propertyID; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$mapped[ $key ] = array( $value );
```
Comment (eugene-manuilov) [inline includes/Modules/Analytics_4.php:633]: There might be more than one web data stream per property. Create a new GA4 account with 2 or more GA4 properties and many web data streams per property. Then see what this function returns for properties from the new account.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @ivankruchkoff. Mostly looks good, but there are a few things that we need to address before we can merge this PR. Please, take a look.

File: includes/Modules/Analytics_4.php (hunk 2 above)
Comment (ivankruchkoff) [reply includes/Modules/Analytics_4.php]: https://github.com/google/site-kit-wp/pull/3755/commits/7886d760d2ff3456eaa4e8dd101be76d379d8702

File: includes/Modules/Analytics_4.php (hunk 3 above)
Comment (ivankruchkoff) [reply includes/Modules/Analytics_4.php]: https://github.com/google/site-kit-wp/pull/3755/commits/7886d760d2ff3456eaa4e8dd101be76d379d8702

File: includes/Core/REST_API/REST_Routes.php (hunk 1 above)
Comment (ivankruchkoff) [reply includes/Core/REST_API/REST_Routes.php:170]: https://github.com/google/site-kit-wp/pull/3755/commits/47134117766ffd6a887f60ee03f4eccef24f4f33

**Review by eugene-manuilov — APPROVED** [review]
Thanks, @ivankruchkoff. It looks good to me.

**Pull request merged**